### PR TITLE
framework upgrade to 0.50.0

### PR DESCRIPTION
**Pull request by alyn509**

### CODE REVIEW

**Review by andrei-marinica — DISMISSED** [review]

File: nft-minter-deployer/src/lib.rs [hunk 1]
```diff
@@ -111,7 +110,9 @@ pub trait NftMinterDeployer: factory::FactoryModule + events::EventsModule {
         );
 
         let _: IgnoreValue = self
-            .user_nft_minter_proxy(nft_minter_address)
+            .tx()
+            .to(nft_minter_address)
+            .typed(nft_minter::nft_minter_proxy::NftMinterProxy)
             .unpause_endpoint()
             .execute_on_dest_context();
```
Comment (BiancaIalangi) [inline nft-minter-deployer/src/lib.rs]: You can replace with ".sync_call_same_context" and remove the assignment to IgnoreValue type

File: royalties-handler/src/nft_minter_interactor.rs [hunk 2]
```diff
@@ -40,14 +42,13 @@ pub trait NftMinterInteractorModule:
         &self,
         sc_address: ManagedAddress,
     ) -> EgldValuePaymentsVecPair<Self::Api> {
-        self.nft_minter_proxy(sc_address)
+        self.tx()
+            .to(sc_address)
+            .typed(nft_minter::nft_minter_proxy::NftMinterProxy)
             .claim_mint_payments()
             .execute_on_dest_context()
```
Comment (BiancaIalangi) [inline royalties-handler/src/nft_minter_interactor.rs]: `.execute_on_dest_context` can be replaced with `.returns(ReturnsResult).sync_call_same_context()`

File: royalties-handler/src/nft_minter_interactor.rs [hunk 3]
```diff
@@ -31,7 +31,9 @@ pub trait NftMinterInteractorModule:
         &self,
         sc_address: ManagedAddress,
     ) -> EgldValuePaymentsVecPair<Self::Api> {
-        self.nft_minter_proxy(sc_address)
+        self.tx()
+            .to(sc_address)
+            .typed(nft_minter::nft_minter_proxy::NftMinterProxy)
             .claim_royalties()
             .execute_on_dest_context()
```
Comment (mihaicalinluca) [inline royalties-handler/src/nft_minter_interactor.rs]: .sync_call() for unified

File: royalties-handler/src/nft_minter_interactor.rs (hunk 2 above)
Comment (mihaicalinluca) [inline royalties-handler/src/nft_minter_interactor.rs]: .sync_call() for unified

File: royalties-handler/tests/nft_minter_setup/mod.rs [hunk 4]
```diff
@@ -52,14 +50,14 @@ where
...
 
-        // init ESDT System SC mock
-        b_mock.create_sc_account_fixed_address(
-            &Address::from(ESDT_SYSTEM_SC_ADDRESS_ARRAY),
-            &rust_zero,
-            None,
-            esdt_system_sc_mock::contract_obj,
-            "ESDT system SC mock path",
-        );
+        // // init ESDT System SC mock
```
Comment (mihaicalinluca) [inline royalties-handler/tests/nft_minter_setup/mod.rs]: forgotten comment

**Review by BiancaIalangi — APPROVED** [review]

**Review by BiancaIalangi — APPROVED** [review]

**Review by mihaicalinluca — APPROVED** [review]

**Pull request merged**